kernel/include/timeout_q.h

```diff
@@ -25,7 +25,8 @@ static inline void _init_timeout(struct _timeout *t, _timeout_func_t fn)
 	t->dticks = _INACTIVE;
 }
 
-void _add_timeout(struct _timeout *to, _timeout_func_t fn, s32_t ticks);
+void _add_timeout(struct _timeout *to, _timeout_func_t fn,
+		  s32_t ticks, bool realtime);
```

Comment:
I don't like this being new argument being named `realtime` since it has nothing to do with realtime clocks.   It means "adjust `ticks` to take into account unprocessed ticks being held while an in-progress clock announcement loop is iterating over deadlines that were missed".  Maybe call it `plus_remaining` or something that ties it to `announce_remaining`.